core/src/main/java/com/linecorp/armeria/server/VirtualHostBuilder.java

```diff
@@ -591,7 +592,7 @@ VirtualHostBuilder addServiceConfigSetters(ServiceConfigSetters serviceConfigSet
     }
 
     VirtualHostBuilder addRouteDecoratingService(RouteDecoratingService routeDecoratingService) {
-        routeDecoratingServices.add(routeDecoratingService);
+        routeDecoratingServices.add(0, routeDecoratingService);
```

Comment:
- How about using `LinkedList.addFirst(...)`?
- Could you add `The specified decorators are executed in reverse order of the insertion.` to the Javadoc of `ServerBuilder.decorator(...)` and `VirtualHostBuilder.decorator(...)`?

Reply:
> How about using LinkedList.addFirst(...)?

Sound good to me! 👍 

> Could you add The specified decorator(s) is/are executed in reverse order of the insertion. to the Javadoc of ServerBuilder.decorator(...) and VirtualHostBuilder.decorator(...)?

It seems a bit insufficient to add only to the mentioned point. Or is it enough? 🤔
- `{ServerBuilder|VirtualHostBuilder}.routeDecorator(...)`
- `{ServerBuilder|VirtualHostBuilder}.decoratorUnder(...)`

Shouldn't we add it here too? What do you think?

Reply:
It looks good to add to `routeDecorator` and `decoratorUnder` too. :)

Reply:
I fixed 👍